Fixed the MM-33210 issue

manojmalik20 opened the pull request:
Made loadCustomEmojisForCustomStatusesByUserIds and loadCustomEmojisIfNeeded actions
Added the call for loadCustomEmojisForCustomStatusesByUserIds in status actions
Added the call for loadCustomEmojisForCustomStatusesByUserIds in loadProfiles for DM and GM
Added the call for loadCustomEmojisForCustomStatusesByUserIds in loadPosts and loadUnreads
Removed the loading custom emoji logic from custom status emoji component
Added the call for loadCustomEmojisIfNeeded in custom status modal for suggestions
Replaced getStatusesByIds with loadStatusesByIds in more direct channels modal

<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A description of what this pull request does.
-->

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/mattermost-server/issues/XXXXX

Otherwise, link the JIRA ticket.
-->

#### Related Pull Requests
<!--
List all PRs related to resolving a ticket. For instance, if you submitted a PR to `mattermost/mattermost-server`, please include it here.
-->
- Has server changes (please link here)
- Has redux changes (please link here)
- Has mobile changes (please link here)

#### Screenshots
<!--
If the PR includes UI changes, include screenshots/GIFs.
-->

#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates)
* API additions—new endpoint, new response fields, or newly accepted request parameters
* Database changes (any)
* Websocket additions or changes
* Anything noteworthy to a Mattermost instance administrator (err on the side of over-communicating)
* New features and improvements, including behavioural changes, UI changes and CLI changes
* Bug fixes and fixes of previous known issues
* Deprecation warnings, breaking changes, or compatibility notes

If no release notes are required write NONE. Use past-tense. Newlines are stripped.

Examples:

```release-note
Added new API endpoints POST /api/v4/foo, GET api/v4/foo, and GET api/v4/foo/:foo_id.
```

```release-note
Added a new config setting ServiceSettings.FooBar. Added a new column Foo to the Users table.
```

```release-note
NONE
```
-->
```release-note
```

manojmalik20 committed:
Fixed the MM-33210 issue
Made loadCustomEmojisForCustomStatusesByUserIds and loadCustomEmojisIfNeeded actions
Added the call for loadCustomEmojisForCustomStatusesByUserIds in status actions
Added the call for loadCustomEmojisForCustomStatusesByUserIds in loadProfiles for DM and GM
Added the call for loadCustomEmojisForCustomStatusesByUserIds in loadPosts and loadUnreads
Removed the loading custom emoji logic from custom status emoji component
Added the call for loadCustomEmojisIfNeeded in custom status modal for suggestions
Replaced getStatusesByIds with loadStatusesByIds in more direct channels modal

chetanyakan commented on actions/emoji_actions.jsx:
```diff
@@ -72,3 +75,67 @@ export function addRecentEmoji(alias) {
+        const systemEmojis = EmojiIndicesByAlias;
+        const customEmojisByName = getCustomEmojisByName(state);
+        const nonExistentCustomEmoji = state.entities.emojis.nonExistentEmoji;
+        const emojisToLoad = new Set();
```
Do we need to de-duplicate the emojis again here?

manojmalik20 committed:
Code refactoring

chetanyakan commented on actions/views/channel.js:
```diff
@@ -345,6 +356,17 @@ export function loadPosts({channelId, postId, type}) {
+                    userIds.add(post.user_id);
+                }
+            });
+            dispatch(loadCustomEmojisForCustomStatusesByUserIds(userIds));
```
nit - Refactor this to a function `loadCustomStatusEmojisForPostList`

chetanyakan commented on components/custom_status/custom_status_modal.tsx:
```diff
@@ -66,7 +67,16 @@ const CustomStatusModal: React.FC<Props> = (props: Props) => {
+    const loadCustomEmojisForRecentStatuses = () => {
+        const emojisToLoad = new Set<string>();
+        recentCustomStatuses.forEach((customStatus: UserCustomStatus) => emojisToLoad.add(customStatus.emoji));
+        dispatch(loadCustomEmojisIfNeeded(emojisToLoad));
```
Do we need to call `Array.from` here?

manojmalik20 committed:
Fixed the custom emoji not loading for user updated event
Added logic for loading custom emojis in handleUserUpdatedEvent
Refactored loadCustomEmojisIfNeeded in emoji_actions
Updated import paths and order

manojmalik20 committed:
Refactored loading custom emojis for posts into a separate function
